common/scala/src/main/scala/org/apache/openwhisk/core/database/elasticsearch/ElasticSearchActivationStore.scala

```diff
@@ -0,0 +1,482 @@
+
+    require(skip >= 0, "skip should be non negative")
+    require(limit >= 0, "limit should be non negative")
+
+    val start = transid.started(this, LoggingMarkers.DATABASE_QUERY, s"[QUERY] 'activations'")
+    val startRange = rangeQuery("start")
+      .gte(since.map(_.toEpochMilli).getOrElse(minStart))
+      .lte(upto.map(_.toEpochMilli).getOrElse(maxStart))
+    val index = generateIndex(namespace.namespace)
+
+    val res = client
+      .execute {
+        search(index) query { must(nameQuery, startRange) } sortByFieldDesc "start" limit limit from skip
+      }
+      .map { res =>
+        if (res.status == StatusCodes.OK.intValue) {
+          val out =
+            if (includeDocs)
+              Right(res.result.hits.hits.map { hit =>
+                restoreAnnotations(restoreResponse(hit.sourceAsString.parseJson.asJsObject)).convertTo[WhiskActivation]
+              }.toList)
+            else
+              Left(res.result.hits.hits.map { hit =>
+                restoreAnnotations(restoreResponse(hit.sourceAsString.parseJson.asJsObject))
+                  .convertTo[WhiskActivation]
+                  .summaryAsJson
+              }.toList)
+          transid.finished(this, start, s"[QUERY] 'activations' completed: matched ${res.result.hits.total}")
+          out
+
+        } else {
+          transid.failed(this, start, s"Unexpected http response code: ${res.status}", ErrorLevel)
+          throw new Exception("Unexpected http response code: " + res.status)
+        }
+      }
+
+    reportFailure(
+      res,
+      failure =>
+        transid.failed(this, start, s"failed to query activation with error ${failure.getMessage}", ErrorLevel))
+  }
+
+  private def restoreAnnotations(js: JsObject): JsObject = {
+    val annotations = js.fields
+      .get("annotations")
+      .map { anno =>
+        Try {
+          JsArray(anno.asJsObject.fields map { p =>
+            JsObject("key" -> JsString(p._1), "value" -> p._2)
+          } toSeq: _*)
+        }.getOrElse(JsArray.empty)
+      }
+      .getOrElse(JsArray.empty)
+    JsObject(js.fields.updated("annotations", annotations))
+  }
+
+  private def restoreResponse(js: JsObject): JsObject = {
+    val response = js.fields
+      .get("response")
+      .map { res =>
+        val temp = res.asJsObject.fields
+        Try {
+          val result = temp
+            .get("result")
+            .map { r =>
+              val JsString(data) = r
+              data.parseJson.asJsObject
+            }
+            .getOrElse(JsObject.empty)
+          JsObject(temp.updated("result", result))
+        }.getOrElse(JsObject(temp - "result"))
+      }
+      .getOrElse(JsObject.empty)
+    JsObject(js.fields.updated("response", response))
+  }
+
+  private def generateIndex(namespace: String): String = {
+    elasticSearchConfig.indexPattern.dropWhile(_ == '/') format namespace.toLowerCase
```

Comment:
we can provide a custom pattern to store activations in special indices

for example, with pattern "openwhisk-%s", activations in different namespaces will be saved in to different indices

or use "openwhisk" with out a "%s", then all activations will be saved into the "openwhisk" index

we can even use index alias like:
- openwhisk-whisk.system(this is an alias and below are indices it connects to) 
   + openwhisk-whisk.system_201911(write index)(activations will be write to this index only while list/get activations search all indices)
   + openwhisk-whisk.system_201910
   + openwhisk-whisk.system_201909

- openwhisk-ns1
   + openwhisk-ns1_201911(write index)
   + openwhisk-ns1_201910
   + openwhisk-ns1_201909

- openwhisk-ns2(we can use shared indices for namespaces which are not very active)
   + openwhisk@shared_201911(write index)
   + openwhisk@shared_201910
   + openwhisk@shared_201909

- openwhisk-ns3
   + openwhisk@shared_201911(write index)
   + openwhisk@shared_201910
   + openwhisk@shared_201909